ui/cypress/e2e/buckets.test.ts

```diff
@@ -60,28 +74,26 @@ describe('Buckets', () => {
 
     describe('Searching and Sorting', () => {
       it('can sort by name and retention', () => {
-        cy.getByTestID('name-sorter').click()
-        cy.getByTestID('bucket-card')
-          .first()
-          .contains('defbuck')
-
-        cy.getByTestID('name-sorter').click()
-        cy.getByTestID('bucket-card')
-          .first()
-          .contains('_monitoring')
-
-        cy.getByTestID('retention-sorter').click()
-        cy.getByTestID('bucket-card')
-          .first()
-          .contains('_tasks')
-
-        cy.getByTestID('retention-sorter').click()
-        cy.getByTestID('bucket-card')
-          .first()
-          .contains('defbuck')
+        const buckets = ['defbuck', '_tasks', '_monitoring']
+        cy.getByTestID('name-sorter')
+          .click()
+          .then(() => {
+            cy.get('.cf-resource-name').each((val, index) => {
```

Comment:
still have to change this (and change `bucket + name` back to `bucket` on the testid in the component)

Reply:
why is that? We changed it for greater specificity to access a specific bucket card.

Reply:
so use that specificity in the test by generating ids to look up (instead of evaluating the text value). i think we're still straddling two solutions right now and just need to pick one and push it through.

Reply:
@drdelambre i'm not quite sure I understand your solution, would you mind expanding upon that with an example of what you had in mind?

Reply:
yeah. so you have the name of the resource in both the test id and in the text content of the dom node you are trying to pinpoint. So you have two sources of truth, and two directions to go to unify them:
1) revert the test id specificity in the component, and test that the value within the component (with `expect(val.text()).to.include(buckets[index])`) is equal to the resource name. that keeps things scoped to buckets (good) but also ties the representation of the key to the value of the key (bad).
2) don't test for the value of the node, but instead look for the test id that meets the key you are looking for( ```cy.getByTestId(`bucket ${cool name}`)``` ). this scopes it to the bucket (good) as well as decouples the representation from the state (good)

seems like you're leaning towards #2. either one is fine, it's just that generic resource selector has got to go.

Reply:
but wouldn't the second solution defeat the purpose of the test? If we're trying to test the sorting mechanism to present a specific order, targeting a bucket by the name would only validate that the bucket exists. It wouldn't represent the order with which that bucket exists in the context of the page

Reply:
Oh wait, I think I see what you're saying. Is it possible to compare two nodes?